### PR TITLE
fix: #21 使用 player_play_music 接口，兼容部分不能播放的设备

### PR DESCRIPTION
**Pull request by hanxi**
兼容了 _type = 1 时亮灯, _type = 2 时不亮灯。

### CODE REVIEW

**Comment by yihong0618** [comment]
Thanks will take a look tonight

File: miservice/minaservice.py [hunk 1]
```diff
@@ -105,11 +116,47 @@ async def player_set_loop(self, deviceId, type=1):
...
+            }
+        }
+        strmusic = json.dumps(music)
         return await self.ubus_request(
             deviceId,
-            "player_play_url",
+            "player_play_music",
             "mediaplayer",
-            {"url": url, "type": _type, "media": "app_ios"},
+            {"startaudioid": audio_id, "music": f"{strmusic}"},
```
Comment (yihong0618) [inline miservice/minaservice.py:159]: no need to f-string here

File: miservice/minaservice.py [hunk 2]
```diff
@@ -105,11 +116,47 @@ async def player_set_loop(self, deviceId, type=1):
...
+            deviceId = h.get("deviceID", "")
+            hardware = h.get("hardware", "")
+            self.device2hardware[deviceId] = hardware
+
+    async def play_by_music_url(self, deviceId, url, _type):
+        _LOGGER.debug("play_by_music_url url:%s, type:%d", url, _type)
+        audio_type = ""
+        if _type == 1:
+            # If set to MUSIC, the light will be on
+            audio_type = "MUSIC"
```
Comment (yihong0618) [inline miservice/minaservice.py:144]: interesting

File: miservice/minaservice.py [hunk 3]
```diff
@@ -105,11 +116,47 @@ async def player_set_loop(self, deviceId, type=1):
...
+                "mediaplayer",
+                {"url": url, "type": _type, "media": "app_ios"},
+            )
+
+    async def _init_devices(self):
+        hardware_data = await self.device_list()
+        for h in hardware_data:
+            deviceId = h.get("deviceID", "")
+            hardware = h.get("hardware", "")
+            self.device2hardware[deviceId] = hardware
```
Comment (yihong0618) [inline miservice/minaservice.py:137]: if deviceId and hardware:
    self.device2hardware[deviceId] = hardware
    break

File: miservice/minaservice.py (hunk 2 above)
Comment (hanxi) [reply miservice/minaservice.py:145]: Any suggestion?

**Pull request merged**